server/src/main/java/org/opensearch/index/snapshots/blobstore/SlicedInputStream.java

```diff
@@ -100,6 +101,11 @@ public final int read() throws IOException {
 
     @Override
     public final int read(byte[] buffer, int offset, int length) throws IOException {
+        Objects.checkFromIndexSize(offset, length, buffer.length);
+        if (length == 0) {
```

Comment:
should it be <= ?

Reply:
`Objects.checkFromIndexSize` handles the invalid cases, so less than zero will never get here.